pkg/test/framework/components/istio/egress.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istio
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/image"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+const (
+	egressIstioLabel   = "egressgateway"
+	egressServiceName  = "istio-" + egressIstioLabel
+	egressOperatorYAML = `
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  profile: empty
+  components:
+    egressGateways:
+      - name: istio-egressgateway
+        enabled: true
+`
+)
+
+// deployEgress creates and applies a separate deployment for istio-egressgateway.
+func (i *operatorComponent) deployEgress(cluster resource.Cluster) error {
```

Comment:
This could probably be abstracted to `deployGateway(cluster, yaml, label)` but not really a big deal.

Reply:
ack ... we can always refactor in the future if we think it's useful.